pkg/cluster/cluster.go

```diff
@@ -378,6 +384,34 @@ func (c *Cluster) deleteClientServiceLB() error {
 	return nil
 }
 
+func (c *Cluster) deleteUnusedServices() error {
+	s := labels.SelectorFromSet(map[string]string{
+		"etcd_cluster": c.Name,
+		"app":          "etcd",
+	})
+	r, err := labels.NewRequirement("etcd_node", selection.Exists, nil)
+	if err != nil {
+		return err
+	}
+
+	svcs, err := c.KubeCli.Services(c.Namespace).List(k8sapi.ListOptions{
```

Comment:
Ah. I see.
I think the problem is that we only list pods and then call removePodAndService() only on Pod name.
Instead of calling another func to delete services, could you rewrite that part in delete() to list and delete both pods and services?